# Problemas de prueba de hipótesis

## Ejercicio 1

Usted es un nutricionista que investiga dos tipos diferentes de dietas para ver si existe una diferencia significativa en la pérdida de peso después de un mes. Eliges dos grupos aleatorios de personas; un grupo sigue la primera dieta y el otro sigue la segunda. Al final del mes, se registra la pérdida de peso (en kg) de cada persona:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

Con estos datos se busca responder a la siguiente pregunta: ¿Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta?

Para sacar conclusiones, siga los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba para comprobar la hipótesis. Puede utilizar la prueba t de Student.
-Analizar las conclusiones.

In [1]:
from scipy import stats
import pandas as pd

data = {
    "Diet1":[2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5],
    "Diet2":[3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2]
}

df = pd.DataFrame(data)

stats.ttest_ind((df["Diet1"]), (df["Diet2"]))

TtestResult(statistic=np.float64(-3.5383407969933938), pvalue=np.float64(0.004083270191713912), df=np.float64(12.0))

### Hipótesis para el Ejercicio 1

-   **Hipótesis nula ($H_0$)**: No hay una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la Dieta 1 y las que siguieron la Dieta 2. Es decir, $μ_1 = μ_2$.
-   **Hipótesis alternativa ($H_1$)**: Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la Dieta 1 y las que siguieron la Dieta 2. Es decir, $μ_1 \neq μ_2$.

### Análisis de las conclusiones para el Ejercicio 1

El resultado de la prueba t de Student es `TtestResult(statistic=np.float64(-3.5383407969933938), pvalue=np.float64(0.004083270191713912), df=np.float64(12.0))`.

-   **Estadístico t**: `-3.538`
-   **Valor p**: `0.00408`
-   **Grados de libertad (df)**: `12`

Para analizar las conclusiones, comparamos el valor p con un nivel de significancia (α) comúnmente utilizado, por ejemplo, α = 0.05.

Dado que el valor p (0.00408) es menor que el nivel de significancia (0.05), **rechazamos la hipótesis nula**.

Esto significa que hay evidencia estadística suficiente para concluir que **existe una diferencia significativa en la pérdida de peso promedio entre la Dieta 1 y la Dieta 2**.

## Prueba t de Student

La **prueba t de Student** es una prueba estadística inferencial utilizada para determinar si existe una diferencia significativa entre las medias de dos grupos. Se utiliza comúnmente cuando se trabaja con muestras pequeñas y la desviación estándar de la población es desconocida.

### Propósito

El objetivo principal de la prueba t de Student es comparar las medias de dos grupos y determinar si la diferencia observada entre ellas es estadísticamente significativa o si podría haber ocurrido por casualidad. Es decir, nos ayuda a decidir si la hipótesis nula (no hay diferencia entre las medias) debe ser rechazada en favor de la hipótesis alternativa (hay una diferencia).

### Tipos de Prueba t de Student

Existen varias versiones de la prueba t, pero las más comunes son:

1.  **Prueba t para muestras independientes (unpaired t-test)**: Se utiliza cuando se comparan las medias de dos grupos de sujetos diferentes e independientes (como en el Ejercicio 1, donde se comparan dos dietas con grupos de personas diferentes).
2.  **Prueba t para muestras pareadas (paired t-test)**: Se utiliza cuando se comparan las medias de dos mediciones tomadas del mismo grupo de sujetos (por ejemplo, antes y después de un tratamiento).
3.  **Prueba t de una muestra (one-sample t-test)**: Se utiliza para comparar la media de una sola muestra con un valor conocido de la población.

### Supuestos Clave

Para que los resultados de la prueba t de Student sean válidos, generalmente se deben cumplir ciertos supuestos:

*   **Independencia de las observaciones**: Las observaciones dentro de cada grupo y entre los grupos deben ser independientes.
*   **Normalidad**: Los datos en cada grupo deben seguir una distribución aproximadamente normal. Para muestras grandes, el Teorema del Límite Central puede relajar este supuesto.
*   **Homogeneidad de varianzas (para la prueba t de muestras independientes)**: Las varianzas de los dos grupos que se comparan deben ser aproximadamente iguales. Si este supuesto no se cumple, se puede utilizar una versión ajustada de la prueba t (como la prueba t de Welch).

### Interpretación de los Resultados

Los resultados de una prueba t suelen incluir:

*   **Estadístico t**: Es un valor que resume la diferencia entre las medias de los grupos en relación con la variabilidad dentro de los grupos. Un valor t más grande (en valor absoluto) indica una mayor diferencia entre las medias.
*   **Valor p**: Es la probabilidad de obtener un estadístico t tan extremo (o más) como el observado, asumiendo que la hipótesis nula es verdadera (es decir, no hay diferencia real entre las medias). Es el valor más crucial para la toma de decisiones.
*   **Grados de libertad (df)**: Indica el número de piezas de información independientes que están disponibles para estimar la variabilidad en los datos.

Para interpretar el valor p, se compara con un **nivel de significancia (α)** predefinido, que comúnmente es 0.05 (o 5%).

*   **Si p < α**: Se **rechaza la hipótesis nula**. Esto significa que hay suficiente evidencia estadística para concluir que existe una diferencia significativa entre las medias de los grupos.
*   **Si p ≥ α**: Se **falla en rechazar la hipótesis nula**. Esto significa que no hay suficiente evidencia estadística para concluir que existe una diferencia significativa entre las medias. No implica que las medias sean iguales, solo que no hay evidencia para decir que son diferentes.

## ANOVA

**ANOVA** (*Analysis of Variance - Análisis de variación*) Es una técnica estadística utilizada para comparar las medidas de dos o más grupos. La idea detrás de ANOVA es descomponer la variabilidad total de los datos en dos componentes: variabilidad entre grupos y variabilidad dentro de los grupos:

- **Variabilidad entre grupos**: Esta variabilidad se refiere a las diferencias entre las medias del grupo. Si esta variabilidad es considerablemente mayor que la variabilidad dentro del grupo, podría ser una indicación de que al menos una de las medias del grupo es diferente.
- **Variabilidad dentro del grupo**: Esta variabilidad se refiere a la dispersión de los datos dentro de cada grupo. Si todos los grupos tienen una variabilidad similar, entonces cualquier diferencia notable en las medias de los grupos podría considerarse significativa.

Las hipótesis en ANOVA suelen incluir:

- **Hipótesis nula** ($H₀$): Las medias de todos los grupos son iguales.
- **Hipótesis alternativa** ($H₁$): Al menos una de las medias del grupo es diferente.

Si el resultado de la prueba ANOVA es significativo (por ejemplo, un valor p inferior a un umbral como 0,05), esto sugiere que al menos la media de un grupo es diferente.

## Ejercicio 2

Un agricultor decide probar tres tipos diferentes de fertilizantes para determinar si alguno es superior en términos de producción de maíz. El agricultor planta maíz en 15 parcelas idénticas y utiliza los tres fertilizantes (5 parcelas para cada tipo). Al final de la temporada mide el rendimiento de maíz (en kg) de cada parcela, con el siguiente resultado:

| Fertilizante 1 | Fertilizante 2 | Fertilizante 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

Con estos datos busca responder la siguiente pregunta: ¿Existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes?

Para ayudarte, sigue los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba ANOVA.
-Analizar las conclusiones.
-Si un fertilizante es mejor que otro, ¿cómo podemos saberlo?

In [ ]:
import pandas as pd
from scipy import stats

# Datos de rendimiento de maíz para cada fertilizante
data_fertilizer = {
    "Fertilizante 1": [20, 21, 20, 19, 20],
    "Fertilizante 2": [22, 21, 23, 22, 21],
    "Fertilizante 3": [24, 23, 22, 23, 24]
}

df_fertilizer = pd.DataFrame(data_fertilizer)

# Realizar la prueba ANOVA
f_statistic, p_value = stats.f_oneway(df_fertilizer["Fertilizante 1"],
                                      df_fertilizer["Fertilizante 2"],
                                      df_fertilizer["Fertilizante 3"])

print(f"Estadístico F: {f_statistic}")
print(f"Valor p: {p_value}")